.github/workflows/deploy.yml

```
@@ -0,0 +1,120 @@
+      - name: Install Serverless Framework
+        run: npm install -g serverless
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Create .env file
+        if: github.ref == 'refs/heads/${{ matrix.branch }}'
```

actionlint (1.7.8)

if: condition "github.ref == 'refs/heads/${{ matrix.branch }}'" is always evaluated to true because extra characters are around ${{ }}

(if-cond)